internal/auth/wrapper.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Abstract Machines
+// SPDX-License-Identifier: Apache-2.0
+
+package auth
+
+import (
+	"context"
+
+	"github.com/absmach/magistrala"
+	grpcclient "github.com/absmach/magistrala/auth/api/grpc"
+	"github.com/absmach/magistrala/pkg/auth"
+)
+
+type wrapper struct {
+	auth grpcclient.AuthServiceClient
```

Comment:
Since `grpcclient.AuthServiceClient` is already a wrapper of https://github.com/absmach/magistrala/blob/main/auth/api/grpc/client.go#L34, there is no point in in wrapping it twice. Let's move the implementation here, and use `pkg.AuthWrapper` as a dependency instead of the gRPC client from the Auth service.